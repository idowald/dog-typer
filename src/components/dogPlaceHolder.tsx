import * as React from "react";
import { CircularProgress, createStyles } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

// props from parent element
interface OwnProps {
  dogPlaceHolderUrl: string;
  loadingBreed: boolean;
}
const useStyles = makeStyles(() =>
  createStyles({
    hidden: {
      display: "none"
    },
    float: {
      float: "left",
      marginTop: "140px",
      marginLeft: "300px",
      position: "fixed"
    }
  })
);
export const DogPlaceHolder = ({
  dogPlaceHolderUrl,
  loadingBreed
}: OwnProps) => {
  const classes = useStyles();
  return (
    <div>
      <CircularProgress
        className={loadingBreed ? classes.float : classes.hidden}
      />
      <img
        alt="dog place holder"
        id="placeHolderImageId"
        src={dogPlaceHolderUrl}
        height="360px"
        width="640px"
      />
    </div>
  );
};
