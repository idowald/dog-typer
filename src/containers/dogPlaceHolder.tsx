import { connect } from "react-redux";
import { State } from "../store/reducers/reducer";
import { DogPlaceHolder } from "../components/dogPlaceHolder";

export const DogPlaceHolderContainer = connect(
  ({ dog: { dogPlaceHolderUrl, loadingBreed } }: State) => ({
    dogPlaceHolderUrl,
    loadingBreed
  })
)(DogPlaceHolder);
