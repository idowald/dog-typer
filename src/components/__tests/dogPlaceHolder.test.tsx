import React from "react";
import renderer from "react-test-renderer";

import { DogPlaceHolder } from "../dogPlaceHolder";

test("renders dog place holder", () => {
  const rendered = renderer.create(
    <DogPlaceHolder
      dogPlaceHolderUrl={`${process.env.PUBLIC_URL}/dogBreedPlaceholder.png`}
      loadingBreed={false}
    />
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});
