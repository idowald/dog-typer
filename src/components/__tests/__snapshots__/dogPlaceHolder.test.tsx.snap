// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dog place holder 1`] = `
<div>
  <div
    className="MuiCircularProgress-root makeStyles-hidden-1 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
  <img
    alt="dog place holder"
    className="makeStyles-sm-3"
    height="360px"
    id="placeHolderImageId"
    src="/dogBreedPlaceholder.png"
    width="640px"
  />
</div>
`;
